File: C++/Factorial.cpp
#include<iostream>
using namespace std;

int factorial(int number) 
{
  	if(number==0) 
  	{
  		return 1;
  	}
  	else 
  	{
  		return (number * factorial(number-1));
  	}
}


//driver function
int main()
{
  int number;
  
  cout<<"Enter Last Number : ";
  cin>>number;
  cout<<"\nThe factorial of "<<number<<"  is : "<<factorial(number)<<endl;

  return 0;
}

File: Java/Factorial.java
 
import java.util.*;
 class Factorial {

	public static void main(String[] args) 
	{
		Factorial f=new Factorial(); 
		f.fact1();
	}
	void fact1()
	{
		 int i,fact=1,n;
		 System.out.println("Enter Number");
		 Scanner sc= new Scanner(System.in);
		 n=sc.nextInt();
		for( i=1;i<=n;i++)
		{
		fact=fact*i;
		}
		System.out.println(fact);
	}
}